btree/btree.c

```diff
@@ -2704,32 +2701,30 @@ static void lock_op_unlock(struct td *tree)
 	m0_rwlock_write_unlock(&tree->t_lock);
 }
 
-static struct m0_btree_oimpl *level_alloc(int height)
+static void level_alloc(struct m0_btree_oimpl *oi, int height)
 {
-	struct m0_btree_oimpl *oi;
-
-	oi = m0_alloc(sizeof *oi);
-	if (oi == NULL)
-		return NULL;
 	oi->i_level = m0_alloc(height * (sizeof *oi->i_level));
-	if (oi->i_level == NULL) {
-		m0_free(oi);
-		oi = NULL;
-	}
-	return oi;
 }
 
 static void level_cleanup(struct m0_btree_oimpl *oi,
 			  struct m0_be_tx *tx)
 {
+	/**
+	 * This function assumes the thread is unlocked when level_cleanup runs.
+	 * As with current implementation, there is no such need to call
+	 * level_cleanup in lock mode. In future, if there is such need,
+	 * flag needs to be passed as parameter to this function indicating lock
+	 * is aquired or not by thread as this flag will be needed by node_put
+	 * operation to determine if lock is already aquired by thread.
```

Comment:
Replace these lines with:
If ever there arises a need to call level_cleanup() with the lock owned by the calling thread then this routine will need some changes such as accepting a parameter which would tell us if the lock is already taken by this thread.

Reply:
done.